app/services/foreman_ansible/ansible_template_renderer.rb

```diff
@@ -0,0 +1,86 @@
+if defined? ForemanRemoteExecution
+  module ForemanAnsible
+    class AnsibleTemplateRenderer < InputTemplateRenderer
+      alias_method :playbook, :render
+
+      def initialize(job_template,
+                     host = nil,
+                     invocation = nil,
+                     input_values = nil,
+                     preview = false,
+                     templates_stack = [])
+        super(job_template, host, invocation, input_values, preview,
+              templates_stack)
+      end
+
+      # Takes the variables from host and host group parameters and
+      # sets them in the 'Host variables' and 'Group variables' format
+      # in the inventory. For instance:
+      #
+      # host.example.com
+      #
+      # [hostgroup:vars]
+      # parameter=value
+      #
+      # It does not nest group variables, but instead it creates a new group per
+      # host group full title
+      #
+      def inventory
+        inventory = ''
+        inventory << host_variables
+        inventory << group_variables
+        inventory
+      end
+
+      private
+
+      def host_variables
+        host_parameters = build_parameters_hash(Host, target_hosts)
+
+        variables = ''
+        host_parameters.each do |host, parameters|
+          variables += "#{host.name} "
+          parameters.each do |parameter|
+            variables += "#{parameter.name}=#{parameter.value}"
+          end
+          variables += "\n"
+        end
+
+        variables
+      end
+
+      def group_variables
+        group_parameters = build_parameters_hash(Hostgroup, target_hostgroups)
+
+        variables = ''
+        group_parameters.each do |group, parameters|
+          variables += "[#{group.title}]\n"
+          target_hosts.each do |host|
+            variables += "#{host.name}\n"
+          end
+          variables += "[#{group.title}:vars]\n"
+          parameters.each do |parameter|
+            variables += "#{parameter.first}=#{parameter.last} \n"
+          end
+          variables += "\n"
+        end
+
+        variables
+      end
```

Comment:
In foreman_ansible_inventory we currently have all the the params in a foreman_params hash and the hostvars in a foreman hash - this  makes sure we don't collide with any ansible variables like e.g. groups. I wonder if we should do the same to be compatible in the format? This way ansible playbooks could look the same whether we use remote execution or the external inventory script.

We're also prefixing the host group with "foreman_hostgroup_" to have a separate namespace.